com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/helpers/ApplicationColorDetectorTest.java

```diff
@@ -0,0 +1,164 @@
+package com.sap.cloud.lm.sl.cf.process.helpers;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.flowable.variable.api.history.HistoricVariableInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.sap.cloud.lm.sl.cf.core.dao.OperationDao;
+import com.sap.cloud.lm.sl.cf.core.dao.filters.OperationFilter;
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.Phase;
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.helpers.ApplicationColorDetector;
+import com.sap.cloud.lm.sl.cf.web.api.model.Operation;
+import com.sap.cloud.lm.sl.cf.web.api.model.State;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+import com.sap.cloud.lm.sl.common.util.TestUtil;
+import com.sap.cloud.lm.sl.common.util.TestUtil.Expectation;
+
+public class ApplicationColorDetectorTest {
+
+    private static final String GET_ARGUMENTS_SUCCESSFUL_UNDEPLOY_METHOD_NAME = "getArgumentsSuccessfulUndeploy";
+    private static final String GET_ARGUMENTS_UNSUCCESSFUL_UNDEPLOY_METHOD_NAME = "getArgumentsUnsuccessfulUndeploy";
+    private static final String FAKE_CORRELATION_ID = "39170ac4-57ad-11e9-8647-d663bd873d93";
+    private static final String FAKE_MTA_ID = "mta_id";
+    private static final String DEPLOYED_MTA_01_JSON = "deployed-mta-01.json";
```

Comment:
I don't think we really need these strings taken out as variables - we really only use them to pass as test params and they spare only about 6 characters when used as constants but make someone take 1 more step to find the test data.